src/main.rs

```diff
@@ -1,2 +1,169 @@
+// Вам нужно реализовать программу обработки команд для дисплея.
+// На вход пользователь подает:
+// * 2 числа: размер дисплея
+// * 1 число: цвет дисплея по-умолчанию (1 - красный, 2 - зеленый, 3 - синий)
+// * Последовательность команд: набор чисел.
+//
+// Дисплей поддерживает следующие команды:
+// * 1 x y - переместить курсор в позицию x y
+// * 2 colour - перекрасить пиксель в цвет colour
+//
+// Пример входных данных:
+// 4 4
+// 1
+// 1 2 2 2 3
+//иции (2,2) будет перекрашен в синий цвет
+
+// Обновлять состояние дисплея нужно через метод matrix.set_colour(pos_x, pos_y, colour)
+
+// Важно! Обязательна проверка на ошибки. Если пользователь просит переместиться на пиксель за пределами дисплея или ввел неправильный цвет, то вам нужно кинуть панику!
+
+use std::io;
+
+mod matrix;
+use matrix::Matrix;
+
+struct Display {
+    width: u32,
+    height: u32,
+    matrix: Matrix,
+}
+
+fn create_display(max_width: u32, max_height: u32, default_colour: u8) -> Display {
+    // ваш код сюда
+    Display {
+        width: max_width,
+        height: max_height,
+        matrix: Matrix::new(max_width, max_height, default_colour),
+    }
+}
+
+const WRONG_FORMAT: &'static str = "Wrong data format";
+const WRONG_COORDINATE: &'static str = "Wrong data format: missing point coordinate";
+const WRONG_COLOR: &'static str = "Wrong data format: missing point color";
+const WRONG_PARAMETER: &'static str = "Wrong data format: undefined parameter";
+const POINT_NOT_VALID: &'static str = "Point is not valid";
+
+struct Point{
+    coords: Option<(u64, u64)>,
+    color: Option<u64>,
+}
+
+impl Point {
+
+    fn check(&self, display: &Display){
+        if self.coords == None || self.color == None || Point::is_out_side_point(self, display) {
```

Comment:
```suggestion
        if self.coords == None || self.color == None || self.is_out_side_point(display) {
```

Можно использовать вызов методов на объекте с помощью точки. Так наглядней, короче и привычней.

Reply:
исправил